pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -279,12 +279,18 @@ func (r *ReconcileWildFlyServer) Reconcile(request reconcile.Request) (reconcile
 		wildflyServer.Status.ScalingdownPods = numberOfPodsToScaleDown
 		updateWildflyServer = true
 	}
-	// Ensure the pod states are up to date by switching it to active when statefulset size follows the wilflyserver spec
+	// Ensure the pod states are up to date by switching it to active when statefulset comes back in size (scales up)
 	if numberOfPodsToScaleDown <= 0 {
 		for k, v := range wildflyServer.Status.Pods {
-			if v.State != wildflyv1alpha1.PodStateActive {
+			if v.State != wildflyv1alpha1.PodStateActive { // was in scale down processing
 				wildflyServer.Status.Pods[k].State = wildflyv1alpha1.PodStateActive
 				updateWildflyServer = true
+				// the non-active pod may be in the middle of the scale down process, we need to completely refresh it
+				for _, podItem := range podList.Items {
+					if podItem.Name == v.Name {
+						resources.Delete(wildflyServer, r.client, &podItem)
```

Comment:
It's a good point just I think it's not appropriate to change the `podList` here. The list of pods are those which are known to `WildFlyServer` instance. That list is the same even the `Delete` is called. The delete only removes the `Pod` instance but `StatefulSet` immediately starts it back. The list of pod which is expected to be active for `WildFlyServer` remains unchanged. The `Delete` works as a way to restart/refrest the `Pod` instance setup - when recovery was in processing then the container setup was changed  - it could be in different ways depending what stage of the processing was hit by scale up.
Is the explanation of the purpose enough to have it as it is or did I miss some point here?

Reply:
Thank @ochaloup , we should be fine then.

I was thinking about the possibility when the `StatefulSet` immediately doesn't start the pod, in that case, the `WildFlyServer` status will continue using obsolete information, for example, the pod IP. If the pod is deleted, its IP will change. 

In any case, the `podList` is read again at the beginning of the reconcile loop, if the pending pod is not started yet, the loop will wait at the beginning, and once it is available the `podList` will contain the correct data, which means rest of the code will be using real information.

`WildFlyServer` status will be updated at the end of the loop with the correct information again. So, it would be a corner case where the `WildFlyServer` status could take more time to show the real information. It should not be a problem.

Reply:
Yes, the WildFly server is always "eventually consistent" - e.g. pod could be just deleted manually by a user.